src/benchmarklib/abstract_benchmark_item_runner.cpp

```diff
@@ -22,7 +23,12 @@ std::pair<std::vector<SQLPipelineMetrics>, bool> AbstractBenchmarkItemRunner::ex
     visualize_prefix = std::move(name);
   }
 
-  BenchmarkSQLExecutor sql_executor(_config->enable_jit, _sqlite_wrapper, visualize_prefix);
+  std::shared_ptr<const Table> expected_result_table = nullptr;
+  if (item_id < _dedicated_expected_results.size()) {
```

Comment:
Is `_dedicated_expected_results.size() > 0` but `<= item_id` a valid case or would this be more speaking?
```
if (!_dedicated_expected_results.empty() {
  expected_result_table = _dedicated_expected_results.at(item_id);
```

Reply:
In the current code base, your suggestion would also be possible since currently only the file based item runner implements behavior for `load_dedicated_expected_results`. But your suggested change would integrate a stronger restriction concerning the _dedicated_expected_results vector. This would then always have to have (max_item_id+1)-many elements.

Reply:
I am not sure if you are describing a dense vector (holdings results for [0, 1, 3]) or a sparse vector (holding results for [0, 1, x, 3] with the third slot being empty). However, it should be one of these two, not a semi-sparse vector, where an item_id can either (a) have a dedicated result, (b) fall into the "x" slot, or (c) be bigger than the vector.

If this is unclear, maybe you could stop by. I'm having a hard time describing this.

Reply:
Got it.